### ISSUE TITLE
Proxy setup for Apache

### ISSUE THREAD

**Issue by wi1k1n** [mikecao/umami#237]
Hi. I would like to hide umami behind my current apache2 setup. I've found example config for nginx [in documentation](https://umami.is/docs/running-on-digitalocean) and tried to create similar config for apache. https://github.com/mikecao/umami/issues/134#issuecomment-690962769 seems to answer my question, but I still didn't manage to make it work.

I run umami as it is stated in getting started section so it is accessible on http://example.com:3000.
Main web server is apache which listens to https, so my website is accessible on https://example.com.
What I want is to get umami accessible under https://example.com/umami/.

As far as I understood I need to config apache virtual host to proxy /umami/ requests to localhost:3000 (since it is the same machine). In my example.com.ssl.conf I have the following virtualhost:
```
<VirtualHost *:443>
    ServerName example.com
    DocumentRoot /var/www/example.com/public_html

    # SSL Certificates

    SSLProxyEngine on

    # Proxy for Umami
    <Location /umami>
            ProxyPreserveHost On
            ProxyPass http://localhost:3000/
            ProxyPassReverse http://localhost:3000/
    </Location>
</VirtualHost>
```
So when I request https://example.com/umami/ I get correct html page (so the proxy seems to be working), but then the browser starts to load styles and scripts and it requests them from https://example.com/ (e.g. https://example.com/_next/static/css/b2faf8739526de782384.css) and of course gets 404, since they also have to be proxied from /umami/.

I've spent some time playing around with different proxy directories (tried to set request headers 'Host' and 'X-Forwarded-For' or even tried to use 'ProxyHTMLURLMap' directory to substitute urls in returning html), but unfortunatelly didn't succeed.

I've also found this issue https://github.com/mikecao/umami/issues/13 but if I understood correctly it solves a little different problem.

**Comment by mikecao** [comment]
Try this, https://nextjs.org/docs/api-reference/next.config.js/basepath

**Comment by wi1k1n** [comment]
@mikecao thank you for a quick response! I've checked your reference and it does not seem to work for me. However, I might be doing something wrong.

What I've done is the following:
1. Added _"basepath: '/umami'"_ entry to the 'next.config.js' file (to the root of module.exports dict)
2. Ran umami in developer's mode using 'npm run dev'
3. Checked that project has been recompiled
4. Tried to access the page

What I've observed:
(1) If I access _http://example.com:3000_ (i.e. exactly to the server without proxying) then I receive 404 not found (as expected).
(2) If I access _http://example.com:3000/umami_ then I get correct responses (main page responds with 200, scripts and styles are also loaded correctly). However, after everything is loaded it redirects me to _http://example.com:3000/login_ which shows 404 not found (although the scripts on this _/login_ page are pulled up with correct **:3000/umami/*** basePath)

With proxy this setup behaves as follows:
If I proxy 'https://example.com/umami' to 'http://example.com:3000/' with the following config:
```
<Location /umami>
        ProxyPreserveHost On
        ProxyPass http://localhost:3000/
        ProxyPassReverse http://localhost:3000/
</Location>
```
then https://example.com/umami returns 404 not found (as expected, since this is an equivalent to (1) when checked without proxy).

If I change config as follows:
```
<Location /umami>
        ProxyPreserveHost On
        ProxyPass http://localhost:3000/umami/
        ProxyPassReverse http://localhost:3000/umami/
</Location>
```
then when requesting https://example.com/umami I receive 308 permanent redirect several times and then it hits the limit and stops (so /umami recursively redirects to itself)


From that above I would say that I do not understand the behaviour of (2) and this recursive redirection.

(By the way, I've also found [runtime configuration](https://nextjs.org/docs/api-reference/next.config.js/runtime-configuration) options for next.js, but I haven't found any references in code to this, so correct me please, if I should have used these instead of simple [basePath](https://nextjs.org/docs/api-reference/next.config.js/basepath))

**Comment by mikecao** [comment]
I think the issue is the redirect is happening on the server side, which is not aware of `basePath`. I can update it to do it on the client side instead. I think you should be able to go to `http://example.com/umami/login` manually.

Is there a reason why you're not using a subdomain instead?

**Comment by mikecao** [comment]
Try the latest release. I updated all the code to check for `basePath`.

**Comment by wi1k1n** [comment]
@mikecao thanks for a very quick fix! I've tried the latest release and now it works really well.

To make it work from behind the proxy I also had to fix my apache config. It turned out that the trailing slash (in _ProxyPass_ and _ProxyPassReverse_ directives) did a bad thing. The config that finally worked is:
```
<Location /umami>
        ProxyPreserveHost On
        ProxyPass http://localhost:3000/umami
        ProxyPassReverse http://localhost:3000/umami
</Location>
```
together with `basePath: '/umami'` entry in _next.config.js_


> Is there a reason why you're not using a subdomain instead?

I'm using the free ddns domain which does not allow subdomain entries (or I might be wrong and simply didn't understand how to add those).
